src/webview/components/memory-table.tsx

```diff
@@ -20,18 +20,87 @@ import {
     VSCodeDataGridRow,
     VSCodeDataGridCell
 } from '@vscode/webview-ui-toolkit/react';
-import { Decoration, Memory, StylableNodeAttributes } from '../utils/view-types';
+import { Decoration, Memory, StylableNodeAttributes, isTrigger } from '../utils/view-types';
 import { toHexStringWithRadixMarker } from '../../common/memory-range';
 import { TableRenderOptions } from '../columns/column-contribution-service';
+import { DebugProtocol } from '@vscode/debugprotocol';
 
+export interface MoreMemorySelectProps {
+    count: number;
+    offset: number;
+    options: number[];
+    direction: 'above' | 'below';
+    fetchMemory(partialOptions?: Partial<DebugProtocol.ReadMemoryArguments>): Promise<void>;
+
+}
+
+export const MoreMemorySelect: React.FC<MoreMemorySelectProps> = ({ count, offset, options, fetchMemory, direction }) => {
+    const [numBytes, setNumBytes] = React.useState<number>(options[0]);
+    const containerRef = React.createRef<HTMLDivElement>();
+    const onSelectChange = (e: React.ChangeEvent<HTMLSelectElement>): void => {
+        e.stopPropagation();
+        const { value } = e.currentTarget;
+        setNumBytes(parseInt(value));
+    };
+
+    const loadMoreMemory = (e: React.MouseEvent | React.KeyboardEvent): void => {
+        containerRef.current?.blur();
+        if (isTrigger(e)) {
+            let newOffset = offset;
+            let newCount = count;
+            if (direction === 'above') {
+                newOffset = offset - numBytes;
+                newCount = count + numBytes;
+            } else {
+                newCount = count + numBytes;
+            }
+            fetchMemory({ offset: newOffset, count: newCount });
```

Comment:
Would it be possible to fetch only the new memory and concatenate it with the existing memory?

Reply:
Given uncertainty about exactly what the adapter will have returned given existing parameters, this is more challenging than it sounds.